File: src/fiefapp.svelte/src/models/market.ts
import type { Development } from './industries/development';

interface IMarket {
  id: string;
  stewardId?: string;
  merchantIds: string[];
  soldierIds: string[];
  name: string;
  developmentLevel: number;
  merchandise: number;
  silver: number;
  base: number;
  luxury: number;
  taxes: number;
  bailiffs: number;
  crime: number;
  development?: Development;
}

export class Market implements IMarket {
  id: string;
  stewardId?: string;
  merchantIds: string[];
  soldierIds: string[];
  name: string;
  developmentLevel: number;
  merchandise: number;
  silver: number;
  base: number;
  luxury: number;
  taxes: number;
  bailiffs: number;
  crime: number;
  development?: Development;

  constructor() {
    this.id = '';
    this.stewardId = undefined;
    this.merchantIds = [];
    this.soldierIds = [];
    this.name = '';
    this.developmentLevel = 1;
    this.merchandise = 0;
    this.silver = 0;
    this.base = 0;
    this.luxury = 0;
    this.taxes = 0;
    this.bailiffs = 0;
    this.crime = 0;
    this.development = undefined;
  }
}

File: src/fiefapp.entities/Market.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Market : Industry, IBaseEntity, IFiefForeinKey, ICanHaveSoldiers, ICanBeDeveloped, INeedSteward, IIncomeListItem
    {
        public Market()
        {
            Id = Guid.NewGuid();
            FiefId = Guid.Empty;
            Merchants = new HashSet<Merchant>();
            Soldiers = new HashSet<Soldier>();
        }

        public override Guid Id { get; set; }
        public Guid FiefId { get; set; }
        public override string Name { get; set; }
        public override int DevelopmentLevel { get; set; }
        public override bool IsInformationItem { get; set; }
        public override bool IsBeingDeveloped { get; set; }
        public override int? Wood { get; set; }
        public override int? Stone { get; set; }
        public override int? Iron { get; set; }
        public override int? Quality { get; set; }
        public override int? Silver { get; set; }
        public override int? Base { get; set; }
        public override int? Luxury { get; set; }
        public int Merchandise { get; set; }
        public int Taxes { get; set; }
        public int Bailiffs { get; set; }
        public int Crime { get; set; }
        public override Guid? StewardId { get; set; }
        public override Steward Steward { get; set; }
        public override Guid? DevelopmentId { get; set; }
        public override Development Development { get; set; }
        public ICollection<Guid> MerchantIds { get; set; }
        public ICollection<Merchant> Merchants { get; set; }
        public ICollection<Guid> SoldierIds { get; set; }
        public ICollection<Soldier> Soldiers { get; set; }
    }
}
